website/client/src/components/auth/register.vue

```diff
@@ -0,0 +1,229 @@
+<template>
+  <form
+    @submit.prevent.stop="register"
+  >
+    <div class="form-group">
+      <label
+        v-if="showFormLabel"
+        v-once
+        for="usernameInput">
+          {{ $t('username') }}
+      </label>
+      <input
+        id="usernameInput"
+        v-model="username"
+        class="form-control input-with-error"
+        type="text"
+        :placeholder="$t(showFormLabel?'usernamePlaceholder':'username')"
+        :class="{'input-valid': usernameValid, 'input-invalid': usernameInvalid}"
+      >
+      <div v-for="issue in usernameIssues" :key="issue" class="input-error">{{ issue }}</div>
+    </div>
+    <div class="form-group">
+      <label v-if="showFormLabel" v-once for="emailInput">{{ $t('email') }}</label>
+      <input
+        id="emailInput"
+        v-model="email"
+        class="form-control"
+        type="email"
+        :placeholder="$t(showFormLabel?'emailPlaceholder':'email')"
+        :class="{'input-invalid': emailInvalid, 'input-valid': emailValid}"
+      >
+    </div>
+    <div class="form-group">
+      <label v-if="showFormLabel" v-once for="passwordInput">{{ $t('password') }}</label>
+      <input
+        id="passwordInput"
+        v-model="password"
+        class="form-control"
+        type="password"
+        :placeholder="$t(showFormLabel?'passwordPlaceholder':'password')"
+        :class="{
+            'input-invalid input-with-error': passwordInvalid,
+            'input-valid': passwordValid
+          }"
+      >
+      <div v-if="passwordInvalid" class="input-error">{{ $t('minPasswordLength') }}</div>
+    </div>
+    <div class="form-group">
+      <label
+        v-if="showFormLabel"
+        v-once
+        for="confirmPasswordInput">
+          {{ $t('confirmPassword') }}
+      </label>
+      <input
+        id="confirmPasswordInput"
+        v-model="passwordConfirm"
+        class="form-control input-with-error"
+        type="password"
+        :placeholder="$t(showFormLabel?'confirmPasswordPlaceholder':'confirmPassword')"
+        :class="{'input-invalid': passwordConfirmInvalid, 'input-valid': passwordConfirmValid}"
+      >
+      <div v-if="passwordConfirmInvalid" class="input-error">
+        {{ $t('passwordConfirmationMatch') }}
+      </div>
+      <p v-once class="form-text small" v-html="$t('termsAndAgreement')">
+      </p>
+    </div>
+    <div class="text-center">
+      <button
+        type="submit"
+        class="btn btn-block btn-info sign-up"
+        :disabled="signupFormInvalid"
+        @click="register"
+      >{{ $t('signup') }}</button>
+      <div class="toggle-links" v-if="showToggleLoginLink">
+        <router-link :to="{name: 'login'}" exact="exact">
+          <a v-once class="toggle-link" v-html="$t('alreadyHaveAccountLogin')"></a>
+        </router-link>
+      </div>
+    </div>
+  </form>
+</template>
+
+<style lang="scss" scoped>
+  @import '~@/assets/scss/colors.scss';
```

Comment:
I think this block can be removed as it simply imports color variables but they're not used then